---
title: C26111
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
author: corob-msft
ms.author: corob
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: fb9329661f57c9b6f396c920c5cb8c78457e2bdb
ms.sourcegitcommit: 68f893f6e472df46f323db34a13a7034dccad25a
ms.translationtype: MT
ms.contentlocale: es-ES
ms.lasthandoff: 02/15/2020
ms.locfileid: "77270818"
---
# <a name="c26111"></a>C26111
ADVERTENCIA C26111: el autor de la llamada no puede liberar el bloqueo \<> antes de llamar a la función \<Func >.

 La anotación `_Requires_lock_not_held_` impone una condición previa de que el recuento de bloqueos del bloqueo especificado no pueda ser mayor que cero cuando se llama a la función. Warning C26111 se emite cuando una función no puede liberar el bloqueo antes de llamar a otra función.

## <a name="example"></a>Ejemplo
 En el ejemplo siguiente se genera la advertencia C26111 porque la llamada a `DoNotLock` dentro de la sección bloqueada infringe la condición previa `_Requires_lock_not_held_`.

```cpp
typedef struct _DATA
{
    CRITICAL_SECTION cs;
    int d;
} DATA;

_Requires_lock_not_held_(p->cs)

void DoNotLock(DATA* p)
{
    EnterCriticalSection(&p->cs);
    p->d = 0;
    LeaveCriticalSection(&p->cs);
}

void LockedFunction(DATA* p)
{
    EnterCriticalSection(&p->cs);
    DoNotLock(p); // Warning C26111
    LeaveCriticalSection(&p->cs);
}
```
